Step2 : Deck Finish

hch-ang opened the pull request:
Deck Finish

kakao-david-window committed:
create please....swift

kakao-david-window committed:
???

kakao-david-window committed:
Deck Finish

godrm commented:
우선 슬랙에 올려준 것처럼 커밋 메시지도 작업 단위를 표현하기 때문에 작성하는 방법을 확인해보세요. 
커밋 로그도 읽을 수 있어야 합니다.

godrm commented on PockerApp/PockerApp/Card.swift:
```diff
@@ -7,7 +7,7 @@
+public class Card {
     // 포커에서의 모양은 4개가 있는데, 유니코드를 직접 쓰기엔 복잡하므로 enum을 활용하여 raw type을 String으로 선언하였다. char가 아니라 string인 이유는 정보를 출력할 때 편하게 문자열로 합치기 위해서이다.
     // 조커를 넣어주기 위해 조커도 Figure에 추가해 주었다.
     enum Figure : String {
```
모든 설명을 코드에 남길 필요는 없습니다. 커밋 로그에 남겨도 좋은 내용이 있을 수도 있습니다.

godrm commented on PockerApp/PockerApp/CardDeck.swift:
```diff
@@ -37,4 +45,12 @@ class CardDeck {
+        for card in cards {
+            print(card.description(), terminator:" ")
+        }
+        print("")
```
출력하는 로직은 계산-처리하는 로직과 분리하세요. CardDeck 같은 객체 내부에서 print 하는 것은 지양하세요.

godrm commented on PockerApp/PockerApp/branch_step2.swift:
```diff
@@ -0,0 +1,8 @@
+//  Created by Hochang Lee on 2021/01/07.
+//
+
+import Foundation
```
이 파일은 뭘까요? 커밋할 때는 꼭 필요하고 완성된 파일만 올리세요. 
git add 할 때도 모든 파일을 추가하지 않도록 하세요.

godrm commented on PockerApp/PockerApp/main.swift:
```diff
@@ -13,10 +13,20 @@ public func testFunction() -> Int{
+//var deck = CardDeck()
+//deck.showAll()
+//deck.shuffle()
+//deck.showAll()
```
의미없는 주석은 지우고 커밋하세요.

godrm commented on PockerApp/TestToolsTest/TestToolsTest.swift:
```diff
@@ -28,8 +29,44 @@ class TestToolsTest: XCTestCase {
+
+        XCTAssertEqual(c1.description(), "❤️Q")
+        XCTAssertEqual(c2.description(), "♠7")
+        XCTAssertEqual(c3.description(), "🃏")
```
테스트 함수 이름과 Assert는 같은 맥락으로 이어져 있습니다.
더 구체적인 이름과 내용으로 작성하는 게 좋습니다. 만약 이 테스트가 실패한다면 어느 코드가 잘못된건지 유추할 수 있을까요?

godrm commented on PockerApp/TestToolsTest/TestToolsTest.swift:
```diff
@@ -28,8 +29,44 @@ class TestToolsTest: XCTestCase {
+        
+        // 카드를 리셋한 후 개수가 모두 채워졌는지 체크
+        cDeck.reset()
+        XCTAssertEqual(cDeck.count(), 53)
```
여기 있는 테스트는 단위 테스트라기 보다는 통합 테스트에 가깝습니다. 함수 하나, 기능 하나, 역할 하나만 테스트하도록 분리해보세요.
이 중에 하나가 실패하면 테스트 함수가 실패합니다. 그럼 어떤 Assert인지 비교하는게 아닙니다. 바로 어느 코드가 실패인지 찾을 수 있을까요?

godrm reviewed (commented):
다음과 같은 피드백을 확인해보세요.
디테일한 부분들도 개선해보세요.

kakao-david-window committed:
git ignore test

kakao-david-window committed:
Modify as the feedback I got

erase some unnecessary comments
separate some print codes from the method in the class treating logical process
delete unnecessary file
synchronize naming between test functions and their assert methods
divide a large test into small tests
